File: backend/src/main/java/com/banner_management/backend/service/WebsiteService.java
package com.banner_management.backend.service;

import com.banner_management.backend.entity.WebsiteEntity;
import com.banner_management.backend.repository.WebsiteRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.NoSuchElementException;

@Service
public class WebsiteService {

    @Autowired
    private WebsiteRepository websiteRepository;


    public List<WebsiteEntity> getAllSections(){
        return websiteRepository.findAll();
    }




    public List<WebsiteEntity> listWebsiteByUserAdd(String userAdd) {
        return websiteRepository.getWebsiteEntitiesByUserAdd(userAdd);
    }

    public WebsiteEntity getWebNameByID(int websiteID){
        return websiteRepository.getById(websiteID);
    }

    public Page<WebsiteEntity> getSectionPage(int number){
        PagingAndSortingRepository<WebsiteEntity, Integer> websiteRepo = websiteRepository;
        Page<WebsiteEntity> websites = ((WebsiteRepository) websiteRepo).getWebsiteByPage(PageRequest.of(number, 5));
        return websites;
    }

    public Page<WebsiteEntity> getSectionByPageAndUserAdd(String userAdd, int number){
        PagingAndSortingRepository<WebsiteEntity, Integer> websiteRepo = websiteRepository;
        Page<WebsiteEntity> websites = ((WebsiteRepository) websiteRepo).getWebsiteByPageAndUserAdd(userAdd, PageRequest.of(number, 5));
        return websites;
    }


    @Transactional
    public void save(WebsiteEntity websiteEntity) {
        websiteRepository.save(websiteEntity);
    }

    public WebsiteEntity getById(Integer id){
        return websiteRepository.findById(id).get();
    }

    @Transactional
    public void delete(Integer id){
        try {
            websiteRepository.deleteById(id);
        }catch (NoSuchElementException e){

        }
    }
}

File: frontend/src/services/website/WebsiteService.js
import axios from "axios";

const WEBSITE_API_SECTION_URL = "http://localhost:8080/api/websites";

class WebsiteService {
    getWebsiteByPageAndUserAdd(userAdd, number){
        return axios.get(WEBSITE_API_SECTION_URL + '/page/user=' + userAdd + '/' + number);
    }

    getWebsiteByPage(number){
        return axios.get(WEBSITE_API_SECTION_URL + '/page/' + number);
    }

    createWebsite(websiteItem) {
        return axios.post(WEBSITE_API_SECTION_URL, websiteItem);
    }

    getWebsiteByUserAdd(userAdd){
        return axios.get(WEBSITE_API_SECTION_URL + '/user=' + userAdd);
    }
}

export default new WebsiteService();